packages/components/text-field/src/text-field.ts

```diff
@@ -152,6 +157,17 @@ export class TextField<T extends { toString(): string } = string>
     }
 
     this.setFormControlElement(this.input);
+
+    // This is a workaround, because :has is not working in Safari and Firefox with :host element as it works in Chrome
+    const style = document.createElement('style');
+    style.innerHTML = `
+       sl-number-field:has(input:hover):not(:focus-within),
```

Comment:
Hmm, this is inside the `<sl-text-field>` component. It should not style other components. That is asking for hard-to-debug-issues ;) Can you move the number-field and search-field styling to those components plz?